hamcrest/Hamcrest/Arrays/SeriesMatchingOnce.php

```diff
@@ -31,7 +31,7 @@ public function matches($item)
     public function isFinished()
     {
         if (!empty($this->_elementMatchers)) {
-            $nextMatcher = current($this->_elementMatchers);
+            $nextMatcher = reset($this->_elementMatchers);
```

Comment:
The `reset` moves internal array pointer, but `current` does not. Please add tests first that would show the problem on all PHP versions. Then add a fix.